src/passkey_child/passkey_child_credentials.c

```diff
@@ -90,17 +90,21 @@ prepare_credentials(struct passkey_data *data, fido_dev_t *dev,
 
     DEBUG(SSSDBG_TRACE_FUNC, "Setting user: %s.\n", data->shortname);
 
-    ret = fido_cred_set_user(cred, userid, sizeof(userid), data->shortname,
-                             NULL, NULL);
+    ret = fido_cred_set_user(cred, data->user_id, USER_ID_SIZE,
```

Comment:
Hi,

to be on the safe side it might be better to check if `data->user_id` is not `NULL`. Currently `prepare_credentials()` is only called once but since the memory must be allocated before calling `prepare_credentials()` a check would be good.

bye,
Sumit

Reply:
Done